### PR TITLE
Refactor search form templates

### PR DESCRIPTION
**Pull request by jpalmieri**
@PolRod again, only if you have time.

Remember when we talked about modularizing the templates in this PR https://github.com/jpalmieri/best_search/pull/7 ?

Yep. Here it is.

There are a few pieces to it:
- a `searchForms` object, which contains all of the options for each type of search form
  - e.g., `seachForms.macro` contains a list of options for all of the elements which will appear in that search form
- two templates for each type of row in the search form to be rendered: plain text rows (`search-form-row-text.hdbs`) and date picker rows (`search-form-row-date.hdbs`).
- a template for the finished search form (`search-form-template.hdbs`)
- a function to render the HTML for the search form, using the above templates: `renderSearchForm()`

Here's a summary of how it works:

The user selects a search type from the select list (e.g. "Macros"). The `data-type` attribute from this element is used to pass `searchForms.macros` into the `renderSearchForm()` function. This function renders each row using the `search-form-row` templates, then passes the results of that into `search-form-template.hdbs`, to return the HTML for the completed search form. The return from this function is then rendered on the page, in `<section data-main />`.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -262,6 +266,39 @@
       _getEndDateQuery: function(type) {
```
Comment (PolRod) [inline app.js:266]: This doesn't belong to this PR, but why do you start some elements with "_"?

**Comment by jpalmieri** [comment]
ummm....idk

For some reason I thought it was part of the style guide to start helper functions with an underscore. But I can't figure out why I thought that; I can't find anything that points in that direction. Maybe I saw the underscore functions in someone's code...

I'll remove them. Thanks for pointing that out!

**Comment by PolRod** [comment]
Oh you don't have to remove them! I was just unsure of what was the pattern
there. Do whatever works for ya.
